src/lightning/data/streaming/data_processor.py

```diff
@@ -798,7 +802,6 @@ def _create_process_workers(
             worker = DataWorkerProcess(
```

Comment:
Note: this object takes 15 arguments positionally. This can be very error prone.